test/e2e.test.ts

```diff
@@ -0,0 +1,279 @@
+
+import { fullyExecuteClaim } from "./claiming";
+import { setTime } from "./utils/timeUtils";
+
+interface DeploymentData {
+  cowToken: Contract;
+  vCowToken: Contract;
+  usdcToken: Contract;
+  gnoToken: Contract;
+  wethToken: Contract;
+  deploymentTimestamp: number;
+}
+interface DeploymentParameters {
+  deployer: Wallet;
+  cowDao: Wallet;
+  investorFundsTarget: Wallet;
+  teamController: Wallet;
+  communityFundsTarget: Wallet;
+  provenClaims: ProvenClaims;
+  usdcPrice: BigNumber;
+  gnoPrice: BigNumber;
+  nativeTokenPrice: BigNumber;
+  usdPerCow: BigNumber;
+  initialCowSupply: BigNumber;
+}
+
+async function standardDeployment(
+  deploymentParameters: DeploymentParameters,
+): Promise<DeploymentData> {
+  // Deploy the required tokens
+  const TestERC20 = await ethers.getContractFactory("TestERC20");
+  const usdcToken = await TestERC20.deploy("USDC");
+  const gnoToken = await TestERC20.deploy("GNO");
+  const wethToken = await TestERC20.connect(
+    deploymentParameters.deployer,
+  ).deploy("WETH");
+
+  // Deploying of the CowToken
+  const CowSwapToken = await ethers.getContractFactory(ContractName.RealToken);
+  const realTokenDeploymentParams: RealTokenDeployParams = {
+    cowDao: deploymentParameters.cowDao.address,
+    totalSupply: deploymentParameters.initialCowSupply,
+  };
+  const cowToken = await CowSwapToken.deploy(
+    ...constructorInput(ContractName.RealToken, realTokenDeploymentParams),
+  );
+
+  // Deploying of the CowSwapVirtualToken
+  const CowSwapVirtualToken = await ethers.getContractFactory(
+    ContractName.VirtualToken,
+  );
+  const deploymentParams: VirtualTokenDeployParams = {
+    merkleRoot: deploymentParameters.provenClaims.merkleRoot,
+    realToken: cowToken.address,
+    investorFundsTarget: deploymentParameters.investorFundsTarget.address,
+    gnoToken: gnoToken.address,
+    usdcToken: usdcToken.address,
+    wrappedNativeToken: wethToken.address,
+    communityFundsTarget: deploymentParameters.communityFundsTarget.address,
+    usdcPrice: deploymentParameters.usdcPrice,
+    gnoPrice: deploymentParameters.gnoPrice,
+    nativeTokenPrice: deploymentParameters.nativeTokenPrice,
+    teamController: deploymentParameters.teamController.address,
+  };
+  const vCowToken = await CowSwapVirtualToken.deploy(
+    ...constructorInput(ContractName.VirtualToken, deploymentParams),
+  );
+  const deploymentTimestamp = (await ethers.provider.getBlock("latest"))
+    .timestamp;
+
+  return {
+    cowToken,
+    vCowToken,
+    usdcToken,
+    gnoToken,
+    wethToken,
+    deploymentTimestamp,
+  };
+}
+
+describe("e2e-tests", () => {
+  // In the actual deployment cowDAO, communityFundsTarget, investorFundsTarget, teamController are all gnosis safes.
+  // But to keep the e2e tests simple, we use normal EOA.
+  // This should not make a difference, as in the e2e tests they are only used to send and receive tokens
+  const [
+    deployer,
+    user,
+    user_not_eligible,
```

Comment:
Case for consistency:

```suggestion
    userNotEligible,
```

Weird that linting didn't spot that.